Sources/Testing/Testing.docc/testing-asynchronous-code.md

```diff
@@ -0,0 +1,70 @@
+# Testing asynchronous code
+
+<!--
+This source file is part of the Swift.org open source project
+
+Copyright (c) 2024 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See https://swift.org/LICENSE.txt for license information
+See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+Validate whether your code causes expected events to happen.
+
+## Overview
+
+`swift-testing` integrates with Swift concurrency, meaning that in many
+situations you can test asynchronous code using standard Swift
+features.  Mark your test function as `async` and, in the function
+body, `await` any asynchronous interactions:
+
+```swift
+@Test func priceLookupYieldsExpectedValue() async {
+    #expect(await unitPrice(for: .mozarella) == 3)
```

Comment:
Because `#expect(await ...)` does not expand its arguments, please split this line up:

```suggestion
    let mozarellaPrice = await unitPrice(for: .mozarella)
    #expect(mozarellaPrice == 3)
```

The existing code will compile, but won't produce the best possible output on test failure.